pytket/qir/conversion/conversion.py

```diff
@@ -491,31 +491,55 @@ def circuit_to_module(
             op = command.op
 
             if isinstance(op, RangePredicateOp):
-                lower_qir = pyqir.const(self.qir_int_type, op.lower)
-                upper_qir = pyqir.const(self.qir_int_type, op.upper)
+                # special case handling for REG_EQ
+                if op.lower == op.upper:
+                    registername = command.args[0].reg_name
 
-                registername = command.args[0].reg_name
+                    result = module.module.builder.icmp(
+                        pyqir.IntPredicate.EQ,
+                        pyqir.const(self.qir_int_type, op.lower),
+                        self.ssa_vars[registername],
+                    )
 
-                lower_cond = module.module.builder.icmp(
-                    pyqir.IntPredicate.SGT, lower_qir, self.ssa_vars[registername]
-                )
-                upper_cond = module.module.builder.icmp(
-                    pyqir.IntPredicate.SGT, self.ssa_vars[registername], upper_qir
-                )
+                    condition_bit_index = command.args[-1].index[0]
+                    result_registername = command.args[-1].reg_name
 
-                result = module.module.builder.and_(lower_cond, upper_cond)
+                    self.module.builder.call(
+                        self.set_one_bit_in_reg,
+                        [
+                            self.ssa_vars[result_registername],
+                            pyqir.const(self.qir_int_type, condition_bit_index),
+                            result,
+                        ],
+                    )
 
-                condition_bit_index = command.args[-1].index[0]
-                registername = command.args[-1].reg_name
+                else:
```

Comment:
Github messed the commit up a bit showing what is old and what is new. All of old part got moved here in the else area and the part in the if case is new.